projectforum/projects/project_filters.py

```diff
@@ -5,40 +5,105 @@
 from projectforum.projects.models import Project
 
 
-def get_project_list(status, order, salary, ascending, starting_from,
-                     ending_at):
-    # Sorting is a little more complicated in the case of sort by payment
-    # try:
+def get_projects_by_filter(status, keywords):
+    return projects_JSON_response(filter_projects(status, keywords))
+
+
+def filter_projects(status, keywords):
     project_list = Project.objects.all().filter(status=status)
+    for keyword in keywords:
+        titles = project_list.filter(title__icontains=keyword)
+        descriptions = project_list.filter(description__icontains=keyword)
+        owner_first = project_list.filter(owner__first_name__icontains=keyword)
+        owner_last = project_list.filter(owner__last_name__icontains=keyword)
+        owner = owner_first | owner_last
+        tags = project_list.filter(tags__text__icontains=keyword)
+        project_list = (titles | descriptions | owner | tags).distinct()
+    return project_list
+
+
+def get_project_list(status, keywords, order, salary, ascending):
```

Comment:
Good job seperating everything out into their own functions. This should make testing easier :)